src/main/java/de/craften/ui/swingmaterial/MaterialComboBox.java

```diff
@@ -38,19 +98,23 @@ public void paint(Graphics g) {
         g2.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON);
         g2.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
 
-        g.setColor(Color.WHITE);
+        g.setColor(getBackground());
         g.fillRect(0, 0, getWidth(), getHeight());
 
         g.setFont(Roboto.REGULAR.deriveFont(16f));
-        g2.setColor(Color.BLACK);
+        g.setColor(getSelectedItem() == null ? new Color(getForeground().getRGB() & 0x00FFFFFF | HINT_OPACITY_MASK, true):getForeground());
         FontMetrics metrics = g.getFontMetrics(g.getFont());
-        String text = getSelectedItem() != null ? getSelectedItem().toString() : "";
+        String text = getSelectedItem() != null ? getSelectedItem().toString() : (hint != null ? hint:"");
         g.drawString(text, 0, metrics.getAscent() + (getHeight() - metrics.getHeight()) / 2);
 
         g2.setColor(MaterialColor.GREY_300);
         g2.fillRect(0, getHeight() - 9, getWidth(), 1);
 
-        g2.fillPolygon(new int[]{getWidth() - 5, getWidth() - 10, getWidth() - 15}, new int[]{getHeight() - 24, getHeight() - 19, getHeight() - 24}, 3);
+        if (isFocusOwner()) g2.setColor(accentColor);
```

Comment:
Hmp. I don't see why, but I fail to see why *not* in the same way. Consider it done.